examples/frompapers/Izhikevich_2003.py

```diff
@@ -0,0 +1,179 @@
+# Since there are few neural groups and connections in this simulation, 
+# all the neuron groups and synaptic connections are explicitly coded line by line, 
+# without even using any for loops. 
+
+############################
+
+from brian2 import *
+from brian2tools import *
+import numpy as np
+np.random.seed(1337)
+
+get_ipython().run_line_magic('matplotlib', 'inline')
+defaultclock.dt = 0.5*ms 
+
+# izhikevich model.
+
+eqs = ''' dv/dt = (0.04/ms/mV)*v**2 + (5/ms)*v + 140*mV/ms - u/ms + I*mV/(amp*ms) : volt
+          du/dt = a*(b*v - u)/ms  : volt
+          I : amp
+          a : 1
+          b : 1
+          c : volt
+          d : volt '''
+
+# synaptic parameters are fine-tuned in order to see the activity in the paper.
+# in paper, they used -1*amp for inhibitory and 0.5*amp for excitatory.
+
+we = 2*amp        # weights for excitatory synapses
+wi = -4*amp       # weights for inhibitory synapses
+EL = -65 * mV     # initial value of membrane potentials
+dly = 0*ms        # delay
+
+
+# In[2]:
+
+
+start_scope()
+
+ex_num = 800
+in_num = 200
+
+##define neuron groups
+N_exc = NeuronGroup(ex_num, eqs, threshold='v>30*mV', reset='v=c; u+=d', method='euler')
+N_inh = NeuronGroup(in_num, eqs, threshold='v>30*mV', reset='v=c; u+=d', method='euler')
+
+##define state monitors
+M_exc = StateMonitor(N_exc, ['v', 'u'], record=True)
+M_inh = StateMonitor(N_inh, ['v', 'u'], record=True)
+
+#define spike and rate monitors 
+spikemon_exc = SpikeMonitor(N_exc)
+ratemon_exc = PopulationRateMonitor(N_exc)
+
+spikemon_inh = SpikeMonitor(N_inh)
+ratemon_inh = PopulationRateMonitor(N_inh)
+
+
+# regular spiking for excicatory neurons. gaussian "so-called-noise" is introduced for c and d parameters.
+N_exc.a =0.02
+N_exc.b =0.2
+N_exc.c ='-65*mV + 15*mV*rand()**2'
+N_exc.d = '(8 - 6*rand()**2)*mV'
+
+# fast spiking for inhibitory neurons. noise added to a and b parameters.
+N_inh.a = '(0.02 + 0.08*rand())'
+N_inh.b ='(0.25 - 0.05*rand())'
+N_inh.c = -65*mV
+N_inh.d = 2*mV
+
+##initial values for exc and inh neurons.
+N_exc.v = EL
+N_exc.u = 'b*v'
+
+N_inh.v = EL
+N_inh.u = 'b*v'
+
+
+#defining synaptic connections. weights are randomized.
+
+#from exc to exc.
+Se1 = Synapses(N_exc, N_exc, 'w : amp', delay=dly, on_pre='I_post += w') # excitatory neurons' synapses
+Se1.connect(p=1)  
+Se1.w = 'we*rand()' 
+
+#from inh to exc.
+Se2 = Synapses(N_inh, N_exc, 'w : amp', delay=dly, on_pre='I_post += w') # excitatory neurons' synapses
+Se2.connect(p=1)  
+Se2.w = 'wi*rand()'
+
+#from exc to inh.
+Si1 = Synapses(N_exc, N_inh, 'w : amp', delay=dly, on_pre='I_post += w') # inhibitory neurons' synapses
+Si1.connect(p=1)
+Si1.w = 'we*rand()'
+
+#from inh to inh.
+Si2 = Synapses(N_inh, N_inh, 'w : amp', delay=dly, on_pre='I_post += w') # inhibitory neurons' synapses
+Si2.connect(p=1)
+Si2.w = 'wi*rand()'
+
+
+##input. source: https://www.izhikevich.org/publications/spikes.htm
+N_exc.run_regularly('I = 5*amp*randn()', dt=1*ms) 
```

Comment:
is this supposed to be some white noise random current? if yes, please cf. https://brian2.readthedocs.io/en/stable/user/models.html?highlight=white%20noise#noise

Reply:
Indeed it is. I used run_regularly since the noise is given for every dt=1*ms, not for all the dt (0.5*ms). How should I change it?

Reply:
Here is the paper and MATLAB code: https://www.izhikevich.org/publications/spikes.htm

In code, he updated the 'v' twice at each for loop (a interesting way to do so, and also he did not updated the 'u' variable twice since it is a slow variable and still converges even for 1ms update), while introducing the noise for each dt. So I guess there is a relationship between the dt, gauss, and the single neuron equation.

Reply:
Thanks. I'll have a look asap.

Reply:
@aesagtekin I haven't forgotten. Will do in the next days.

Reply:
Thanks!

Reply:
The `run_regularly` is not correct. All neurons will see the same background current. I haven't found a way to fix it yet.

Reply:
Hi everyone. With the definition `I : amp`, there should be an independent `I` variable per neuron, and therefore `I = 5*amp*randn()` will give a different value for each neuron. The function ["auto-vectorizes"](https://brian2.readthedocs.io/en/stable/advanced/functions.html#functions-with-context-dependent-return-values) according to the context. Or do you have observed that this does not work here?

Reply:
You're right. I had a mistake in my minimal example. So back to square 1 in the comparison with the Matlab implementation.